src/functional/ffront/foast_to_itir.py

```diff
@@ -320,6 +320,23 @@ def visit_BinOp(self, node: foast.BinOp, **kwargs) -> itir.FunCall:
             )
         )
 
+    def visit_TernaryExpr(self, node: foast.TernaryExpr, **kwargs) -> itir.FunCall:
+        lowered_node_cond = self.visit(node.condition, **kwargs)
+        lowered_true_expr = self.visit(node.true_expr, **kwargs)
+        lowered_false_expr = self.visit(node.false_expr, **kwargs)
+
+        # Needed in case of sparse or shifted fields
+        if isinstance(node.type, ct.FieldType) and is_local_kind(node.type):
+            return im.call_("if_")(lowered_node_cond, lowered_true_expr, lowered_false_expr)
+
+        return self._lift_if_field(node)(
+            im.call_("if_")(
+                to_value(node.condition)(lowered_node_cond),
```

Comment:
```suggestion
                lowered_node_cond,
```
Will be a value in any case.